addons/sale_pdf_quote_builder/tests/test_pdf_quote_builder.py

```diff
@@ -144,6 +145,23 @@ def test_dynamic_fields_mapping_for_product_document(self):
             )
             self.assertEqual(result, expected[form_field.name])
 
+    def test_product_document_dialog_params_access(self):
+        sale_order_user_demo = self.sale_order.copy({'user_id': self.user_demo.id})
+        dialog_param = sale_order_user_demo.with_user(
+            self.user_demo.id
+        ).get_update_included_pdf_params()
+        self.assertTrue(dialog_param['headers']['files'])
```

Comment:
In fact, reading the code again, product document could have an empty list in `lines`,  but quotation documents could have an empty list in `files`...
In both case, could you ensure a document is there, and use the check to read the first file's name?